src/pages/ultimate.tsx

```diff
@@ -0,0 +1,256 @@
+    this.state = {
+      available: _.range(9),
+      boards: _.times(9, i => ({
+        cells: _.times(9, j => ({ player: null })),
+      })),
+      xIsNext: true,
+    };
+  }
+
+  handleClick(i: number, j: number) {
+    const boards = this.state.boards.slice();
+    boards[i] = {
+      cells: boards[i].cells.slice(),
+    };
+    if (
+      calculateWinner(boards[i].cells) !== null ||
+      boards[i].cells[j].player !== null
+    ) {
+      return;
+    }
+    boards[i].cells[j].player = this.state.xIsNext
+      ? Winner.Cross
+      : Winner.Nought;
+    let available: number[];
+    if (
+      calculateWinner(
+        boards.map(board => ({ player: calculateWinner(board.cells) })),
+      ) !== null
+    ) {
+      available = [];
+    } else {
+      available =
+        calculateWinner(boards[j].cells) !== null
+          ? _.range(9).filter(
+              board => calculateWinner(boards[board].cells) === null,
+            )
+          : [j];
+    }
+    this.setState(prevState => ({
+      available,
+      boards,
+      xIsNext: !prevState.xIsNext,
+    }));
+  }
+
+  render() {
+    return (
+      <div className={css({ textAlign: 'center' })}>
+        <div className={css({ display: 'inline-block' })}>
+          <LargeBoard
+            available={this.state.available}
+            boards={this.state.boards}
+            handleClick={this.handleClick}
+          />
+        </div>
+      </div>
+    );
+  }
+}
+
+export interface LargeBoardProps {
+  available: number[];
+  boards: Board[];
+  handleClick: (i: number, j: number) => void;
+}
+
+const boardStyle = {
+  display: 'grid',
+  gridTemplateColumns: 'repeat(3, minmax(auto, max-content))',
+};
+
+export const LargeBoard = (props: LargeBoardProps) => (
+  <div
+    className={css({
+      backgroundColor: 'black',
+      border: '1px solid black',
+      gridGap: 1,
+      ...boardStyle,
+    })}
+  >
+    {props.boards.map((board: Board, i: number) => {
+      const availableBoard = props.available.includes(i);
+      return (
+        <SmallBoard
+          key={i}
+          available={availableBoard}
+          board={board}
+          handleClick={j => {
+            if (availableBoard) {
+              props.handleClick(i, j);
+            }
+          }}
+        />
+      );
+    })}
+  </div>
+);
+
+export interface SmallBoardProps {
+  available: boolean;
+  board: Board;
+  handleClick: (i: number) => void;
+}
+
+export const SmallBoard = (props: SmallBoardProps) => (
+  <div
+    className={css({
+      backgroundColor: props.available ? 'green' : 'red',
+      padding: 10,
+      position: 'relative',
+      ...boardStyle,
+    })}
+  >
+    {calculateWinner(props.board.cells) !== null && (
+      <div
+        className={css({
+          height: '100%',
+          left: 0,
+          position: 'absolute',
+          top: 0,
+          width: '100%',
+        })}
+      >
+        {getSymbol(calculateWinner(props.board.cells))()}

```

Comment:
This empty second invocation is a bit odd, but I'm guessing it's here for a reason?